src/dataflow/src/source/kafka.rs

```diff
@@ -374,6 +378,9 @@ impl DataPlaneInfo {
         cp_info: &mut ControlPlaneInfo,
         activator: &Activator,
     ) {
+        // Serve any consumer-global callbacks.
+        self.consumer.poll(Duration::from_secs(0));
```

Comment:
The call to self.consumer.poll was explicitly after the split_partition_queue call because of this comment in the rust library 

`/// `BaseConsumer::poll`. For best results, call `split_partition_queue` before the first call to `BaseConsumer::poll`.
`

Reply:
Oops, my bad for writing some confusing documentation. You need to make sure not to do the following:

```rust
let consumer = Consumer::create();
consumer.subscribe(["foo_topic"]);
consumer.poll();
let pq = consumer.split_partition_queue("foo_topic", 0);
pq.poll(); // will miss some messages
```

This is totally fine because we never call `poll` in between assigning the new partition and splitting off the partition queue. Simplified: 

```rust
loop {
    let consumer = Consumer::create();
    consumer.poll();
    if let Some(pid) = new_partitions_rx.recv() {
        consumer.assign(["foo_topic", pid]);
        queues.push(consumer.split_partition_queue("foo_topic", pid));
    }
}
```